packages/drivers/odsp-driver/src/epochTracker.ts

```diff
@@ -76,15 +77,33 @@ export class EpochTracker implements IPersistedFileCache {
     ): Promise<any> {
         try {
             const value: IVersionedValueWithEpoch = await this.cache.get(this.fileEntryFromEntry(entry));
+            // Version check
             if (value === undefined || value.version !== persistedCacheValueVersion) {
                 return undefined;
             }
+            // Epoch check
             assert(value.fluidEpoch !== undefined, 0x1dc /* "all entries have to have epoch" */);
             if (this._fluidEpoch === undefined) {
                 this.setEpoch(value.fluidEpoch, true, "cache");
             } else if (this._fluidEpoch !== value.fluidEpoch) {
                 return undefined;
             }
+            // Cache expiry check
+            // Only expire the snapshot cache as the ops cache returns ops for a specific range, meaning old ops
+            // will not be retrieved unless they are the "newest" ops.
+            if (entry.type === snapshotKey) {
+                const cacheTime = value.value?.cacheEntryTime;
+                const currentTime = Date.now();
+                if (cacheTime === undefined || currentTime - cacheTime >= defaultCacheExpiryTimeoutMs) {
```

Comment:
When can the cacheTime be undefined? We are always setting the cacheTime when the entry is added to the cache right.

Reply:
Any older caches that still exist before customers switch to the new epochTracker.

Reply:
I see. And we will expire those entries right away. Sound reasonable. Can you add a comment here saying that and also add that to the PR description?